Sources/Build/BuildOperation.swift

```diff
@@ -734,69 +734,41 @@ public final class BuildOperation: PackageStructureDelegate, SPMBuildCore.BuildS
     private func plan(subset: BuildSubset? = nil) throws -> (description: BuildDescription, manifest: LLBuildManifest) {
         // Load the package graph.
         let graph = try getPackageGraph()
-        let buildToolPluginInvocationResults: [ResolvedModule.ID: (target: ResolvedModule, results: [BuildToolPluginInvocationResult])]
-        let prebuildCommandResults: [ResolvedModule.ID: [PrebuildCommandResult]]
-        // Invoke any build tool plugins in the graph to generate prebuild commands and build commands.
-        if let pluginConfiguration, !self.config.shouldSkipBuilding(for: .target) {
+
+        let pluginTools: [ResolvedModule.ID: [String: PluginTool]]
+        // FIXME: This is unfortunate but we need to build plugin tools upfront at the moment because
```

Comment:
Is this really the case? Don't build tool plugin have to declare the inputs and outputs statically compared to pre-build tools which need to be run in order to capture the output files?

Reply:
I am not sure I follow, we need to build tools in order to invoke plugins to form a complete manifest.

Reply:
> Is this really the case? Don't build tool plugin have to declare the inputs and outputs statically compared to pre-build tools which need to be run in order to capture the output files?

IIUC the answer is "no", `createBuildCommands` is invoked during the execution of the plugin, I don't think that could count as "static".